contracts/Profile.sol

```diff
@@ -92,6 +107,46 @@ contract Profile {
     //     profileStorage.setNodeAddress(identityId, hashFunctionId);
     // }
 
+    // TODO: Define where it can be called, change internal modifier
+    function _setAvailableNodeAddresses(uint72 identityId) internal {
+        ProfileStorage ps = profileStorage;
+        HashingProxy hp = hashingProxy;
+
+        UnorderedIndexableContractDynamicSetLib.Contract[] memory hashFunctions = hp.getAllHashFunctions();
+        uint256 hashFunctionsNumber = hashFunctions.length;
+        for (uint8 i; i < hashFunctionsNumber; ) {
+            ps.setNodeAddress(identityId, hashFunctions[i].id);
+            unchecked { i++; }
+        }
+    }
+
+    function stakeReward(uint72 identityId) external onlyAdmin(identityId) {
+        ProfileStorage ps = profileStorage;
+
+        uint96 rewardAmount = ps.getReward(identityId);
+        require(rewardAmount != 0, "You have no reward");
+
+        uint96 oldStake = stakingStorage.totalStakes(identityId);
+        uint96 newStake = oldStake + rewardAmount;
+
+        ps.setReward(identityId, 0);
+        stakingContract.addStake(identityId, rewardAmount);
+
+        if (!shardingTableStorage.nodeExists(identityId) && newStake >= parametersStorage.minimumStake()) {
```

Comment:
is this necessary? I believe the addStake function will handle this part

Reply:
Yes, my bad